core/src/main/scala/runtime/endpoints.scala

```diff
@@ -19,52 +19,92 @@ package opscenter
+import scala.util.Random
 
 object implicits {
 
-  import cats.syntax.applicative._
-
-  implicit def endpointsHandler[M[_]: Applicative]: Endpoints.Handler[M] =
+  implicit def endpointsHandler[M[_]](implicit C: Capture[M]): Endpoints.Handler[M] =
     new Endpoints.Handler[M] {
 
-      def protoMetric: M[HttpService[IO]] =
-        HttpService[IO] {
-          case request @ GET -> Root / "proto" / "models" =>
+      def protoMetricModels: M[HttpService[IO]] =
+        C.capture(HttpService[IO] {
+          case request @ GET -> Root / "proto" / "models" / "metrics" =>
+            StaticFile
+              .fromFile(new File("metrics/src/main/proto/models.proto"), Some(request))
+              .getOrElseF(NotFound())
+        })
+
+      def protoMicroservicesModels: M[HttpService[IO]] =
+        C.capture(HttpService[IO] {
+          case request @ GET -> Root / "proto" / "models" / "microservices" =>
             StaticFile
-              .fromFile(new File("core/src/main/proto/Models.proto"), Some(request))
+              .fromFile(new File("core/src/main/proto/models.proto"), Some(request))
               .getOrElseF(NotFound())
-        }.pure[M]
+        })
 
       def healthcheck: M[HttpService[IO]] =
-        HttpService[IO] {
+        C.capture(HttpService[IO] {
           case GET -> Root / "healthcheck" => Ok(s"Works fine.")
-        }.pure[M]
+        })
 
       def microservices: M[HttpService[IO]] =
-        HttpService[IO] {
+        C.capture(HttpService[IO] {
           case GET -> Root / "microservices" => Ok(getMicroservices.toPB)
-        }.pure[M]
+        })
 
-      def websocketMetrics(
-          metricsStream: Stream[IO, WebSocketFrame],
-          signalFromClient: Sink[IO, WebSocketFrame]): M[HttpService[IO]] =
-        HttpService[IO] {
+      def streamMetrics: M[HttpService[IO]] =
+        C.capture(HttpService[IO] {
           case GET -> Root / "metrics" => WS(metricsStream, signalFromClient)
-        }.pure[M]
+        })
+
+      private def randomMetrics: List[Metric] = {
+        val microservices = List("analytics", "users", "payments")
+        val nodes         = List("node-1", "node-2", "node-4")
+        val metrics       = List("cassandra.queue", "instance.cpu.usage", "instance.cpu.disk")
+
+        for {
+          microservice <- microservices
+          node         <- nodes
+          metric       <- metrics
+          value     = Random.nextInt()
+          timestamp = DateTime.now()
+        } yield Metric(metric, microservice, node, value.toFloat, timestamp)
+
+      }
+
+      private def signalFromClient: Sink[IO, WebSocketFrame] = _.evalMap { (ws: WebSocketFrame) =>
+        ws match {
+          case Text(t, _) => IO.apply(println(s"Received from client: $t"))
+          case f          => IO.apply(println(s"Received from client unknown type: $f"))
+        }
+      }
+
+      private def metricsStream: Stream[IO, WebSocketFrame] =
+        Scheduler[IO](2)
```

Comment:
What's the number 2 mean?

Reply:
That value is the `corePoolSize`. That value is the number of threads to keep in the pool, even if they are idle.

You can see the `apply` here. https://github.com/functional-streams-for-scala/fs2/blob/series/0.10/core/jvm/src/main/scala/fs2/SchedulerPlatform.scala#L13

Reply:
Maybe you could use named parameters clarify it.

Reply:
Make sense. Changed. Thanks for the review buddy.